relay/update_client.go

```diff
@@ -0,0 +1,59 @@
+package relay
+
+import (
+	"context"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/datachainlab/lcp-go/relay/elc"
+)
+
+func updateClient(ctx context.Context, chunkSize uint32, client LCPServiceClient, anyHeader *types.Any, elcClientID string, includeState bool, signer []byte) (*elc.MsgUpdateClientResponse, error) {
+	stream, err := client.UpdateClientStream(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call UpdateClientStream: %w", err)
+	}
+	if err = stream.Send(&elc.MsgUpdateClientStreamChunk{
+		Chunk: &elc.MsgUpdateClientStreamChunk_Init{
+			Init: &elc.UpdateClientStreamInit{
+				ClientId:     elcClientID,
+				IncludeState: includeState,
+				Signer:       signer,
+				TypeUrl:      anyHeader.TypeUrl,
+			},
+		},
+	}); err != nil {
+		return nil, fmt.Errorf("failed to send initial message: %w", err)
+	}
+	chunks, err := splitBytes(anyHeader.Value, chunkSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to call splitBytes: %w", err)
+	}
+	for _, chunk := range chunks {
+		err = stream.Send(&elc.MsgUpdateClientStreamChunk{
+			Chunk: &elc.MsgUpdateClientStreamChunk_HeaderChunk{
+				HeaderChunk: &elc.UpdateClientStreamHeaderChunk{
+					Data: chunk,
+				},
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed send header chunk: %w", err)
```

Comment:
Could you add the chunk index and its size to the error message?

Reply:
Fixed by https://github.com/datachainlab/lcp-go/commit/d8262d6dda4a5416cc52120973f30f55257db41c